diff --git a/scripts/west_commands/runners/bossac.py b/scripts/west_commands/runners/bossac.py
index c29e0dabeba..abc4b4a4999 100644
--- a/scripts/west_commands/runners/bossac.py
+++ b/scripts/west_commands/runners/bossac.py
@@ -25,13 +25,14 @@ class BossacBinaryRunner(ZephyrBinaryRunner):
     '''Runner front-end for bossac.'''
 
     def __init__(self, cfg, bossac='bossac', port=DEFAULT_BOSSAC_PORT,
-                 speed=DEFAULT_BOSSAC_SPEED, boot_delay=0, erase=False):
+                 speed=DEFAULT_BOSSAC_SPEED, boot_delay=0, erase=False, reset=True):
         super().__init__(cfg)
         self.bossac = bossac
         self.port = port
         self.speed = speed
         self.boot_delay = boot_delay
         self.erase = erase
+        self.reset = bool(reset)
 
     @classmethod
     def name(cls):
@@ -39,7 +40,7 @@ class BossacBinaryRunner(ZephyrBinaryRunner):
 
     @classmethod
     def capabilities(cls):
-        return RunnerCaps(commands={'flash'}, erase=True)
+        return RunnerCaps(commands={'flash'}, erase=True, reset=True)
 
     @classmethod
     def do_add_parser(cls, parser):
@@ -56,12 +57,14 @@ class BossacBinaryRunner(ZephyrBinaryRunner):
                             point number) to wait between putting the board
                             into bootloader mode and running bossac;
                             default is no delay''')
+        parser.set_defaults(reset=True)
 
     @classmethod
     def do_create(cls, cfg, args):
         return BossacBinaryRunner(cfg, bossac=args.bossac,
                                   port=args.bossac_port, speed=args.speed,
-                                  boot_delay=args.delay, erase=args.erase)
+                                  boot_delay=args.delay, erase=args.erase,
+                                  reset=args.reset)
 
     def read_help(self):
         """Run bossac --help and return the output as a list of lines"""
@@ -186,8 +189,10 @@ class BossacBinaryRunner(ZephyrBinaryRunner):
 
     def make_bossac_cmd(self):
         self.ensure_output('bin')
-        cmd_flash = [self.bossac, '-p', self.port, '-R', '-w', '-v',
-                     '-b', self.cfg.bin_file]
+        cmd_flash = [self.bossac, '-p', self.port, '-w', '-v', '-b',
+                     self.cfg.bin_file]
+        if self.reset:
+            cmd_flash += ['-R']
 
         if self.erase:
             cmd_flash += ['-e']
