diff --git a/drivers/ethernet/eth_esp32.c b/drivers/ethernet/eth_esp32.c
index 1b202e485..f4f0a8a60 100644
--- a/drivers/ethernet/eth_esp32.c
+++ b/drivers/ethernet/eth_esp32.c
@@ -742,6 +742,7 @@ int eth_esp32_initialize(const struct device *dev)
 	eth_esp32_reset_desc_chain(dev_data);
 	emac_hal_init_mac_default(&dev_data->hal);
 	emac_hal_init_dma_default(&dev_data->hal, &dma_config);
+	emac_ll_pass_all_multicast_enable(dev_data->hal.mac_regs, true);
 
 	/*
 	 * Enable the TX-finish interrupt (for TX descriptor recycling) and
