diff --git a/boards/st/nucleo_f767zi/board.cmake b/boards/st/nucleo_f767zi/board.cmake
index c0b0f91cd26..b693c85d53c 100644
--- a/boards/st/nucleo_f767zi/board.cmake
+++ b/boards/st/nucleo_f767zi/board.cmake
@@ -6,6 +6,6 @@ board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
 board_runner_args(jlink "--device=STM32F767ZI" "--speed=4000")
 
 # keep first
-include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/openocd-stm32.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
