diff --git a/scripts/west_commands/runners/teensy.py b/scripts/west_commands/runners/teensy.py
index dbf093b17ef..1f5814557d6 100644
--- a/scripts/west_commands/runners/teensy.py
+++ b/scripts/west_commands/runners/teensy.py
@@ -7,35 +7,41 @@
 import os
 import subprocess
 
-from runners.core import ZephyrBinaryRunner
+from runners.core import RunnerCaps, ZephyrBinaryRunner
 
 
 class TeensyBinaryRunner(ZephyrBinaryRunner):
     '''Runner front-end for teensy.'''
 
-    def __init__(self, cfg, mcu, teensy_loader):
+    def __init__(self, cfg, mcu, teensy_loader, reset=True):
         super().__init__(cfg)
 
         self.mcu_args = ['--mcu', mcu]
         self.teensy_loader = teensy_loader
         self.hex_name = cfg.hex_file
+        self.reset = bool(reset)
 
     @classmethod
     def name(cls):
         return 'teensy'
 
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'}, reset=True)
+
     @classmethod
     def do_add_parser(cls, parser):
         parser.add_argument('--mcu', required=True,
                             help='Teensy mcu target')
         parser.add_argument('--teensy', default='teensy_loader_cli',
                             help='path to teensy cli tool, default is teensy_loader_cli')
+        parser.set_defaults(reset=True)
 
     @classmethod
     def do_create(cls, cfg, args):
         ret = TeensyBinaryRunner(
             cfg, args.mcu,
-            teensy_loader=args.teensy)
+            teensy_loader=args.teensy, reset=args.reset)
         return ret
 
     def do_run(self, command):
@@ -52,6 +58,7 @@ class TeensyBinaryRunner(ZephyrBinaryRunner):
 
         cmd = ([self.teensy_loader] +
                self.mcu_args +
+               ([] if self.reset else ['-n']) +
                [fname])
 
         self.logger.info(f'Flashing file: {fname}')
